c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -688,6 +831,88 @@ public:
     });
   }
 
+  kj::Promise<void> connect(ConnectContext context) override {
+    auto params = context.getParams();
+    auto host = params.getHost();
+    auto headers = factory.headersToKj(params.getHeaders());
+    auto upPipe = kj::newOneWayPipe();
+
+    PipelineBuilder<ConnectResults> pb;
+    pb.setUp(factory.streamFactory.kjToCapnp(kj::mv(upPipe.out)));
+
+    context.setPipeline(pb.build());
+
+    // This is similar to kj::HttpClientAdapter::connect.
+    auto pipe = kj::newTwoWayPipe();
+    auto paf = kj::newPromiseAndFulfiller<kj::HttpClient::ConnectRequest::Status>();
+    auto response = kj::refcounted<HttpOverCapnpConnectResponseImpl>(kj::mv(paf.fulfiller));
+
+    auto promise = inner->connect(host, headers, *pipe.ends[0], *response)
+        .eagerlyEvaluate([response=kj::mv(response),
+                          host=kj::mv(host),
+                          headers=kj::mv(headers),
+                          connectStream=kj::mv(pipe.ends[0])](kj::Exception&& ex) mutable {
+        response->handleException(kj::mv(ex), kj::mv(connectStream));
+      });
+
+    return paf.promise.then(
+        [this, upPipe = kj::mv(upPipe), down = params.getDown(), context = kj::mv(context),
+        io = kj::mv(pipe.ends[1])](auto status) mutable {
+      auto headers = status.headers->clone();
+
+      if (status.statusCode < 200 || status.statusCode > 299) {
+        auto req = context.getParams().getContext().startErrorRequest();
+        auto rpcResponse = req.initResponse();
+        rpcResponse.setStatusCode(status.statusCode);
+        rpcResponse.setStatusText(status.statusText);
+        rpcResponse.adoptHeaders(factory.headersToCapnp(
+            headers, Orphanage::getForMessageContaining(rpcResponse)));
+
+        // Set the body size if the error body exists.
+        auto bodySize = 0;
+        KJ_IF_MAYBE(errStream, status.errorBody) {
+          KJ_IF_MAYBE(size, errStream->get()->tryGetLength()) {
+            bodySize = *size;
+          }
+        }
+        rpcResponse.getBodySize().setFixed(bodySize);
```

Comment:
Looks like if `tryGetLength()` returns `nullptr` (meaning "unknown"), you are doing `setFixed(0)`, which means the length is known to be zero. You should be doing `getBodySize().setUnknown()`. Or, since "unknown" is the default, you could just move this line into the block above, so that the value is left as "unknown" otherwise. (And then you don't need the `bodySize` local.)